src/printLabels.js

```diff
@@ -111,19 +111,103 @@ function createLabelFileFromSheet(orderDetails) {
 function printLabelFromFile(filename_url) {
   // verify filename exists and we have access to it
   var file = null;
+  var printer = null;
   try {
-    file = DriveApp.getFileById(DocumentApp.openByUrl(filename_url).getId());
+    file = DocumentApp.openByUrl(filename_url)
   } catch (e) {
     Logger.log(e);
     //TODO: why would it ever get to this state, and we should be returning 'false'
     return true;
   }
-  var printSuccessful = false;
+  var printSuccessful = printGoogleDocument(file.getId(), printer, file.getName());
 
   //TODO: replace 'true' with perform print
   if (true) {
     printSuccessful = true;
   }
 
   return printSuccessful;
+}
+
+
+//https://ctrlq.org/code/20061-google-cloud-print-with-apps-script
+function getCloudPrintService() {
+  //TODO: replace CLIENT_ID/SECRET
```

Comment:
We should probably put the resulting Bearer token into the [DocumentCache](https://developers.google.com/apps-script/reference/cache/cache-service#getDocumentCache()) for reuse. My guess is that it won't expire throughout the night and there's no reason to do the OAuth dance for every transaction.